addons/website_event/models/event_event.py

```diff
@@ -560,3 +556,44 @@ def _search_render_results(self, fetch_fields, mapping, icon, limit):
                 end = self.env['ir.qweb.field.date'].record_to_html(event, 'date_end', {})
                 data['range'] = '%s🠖%s' % (begin, end) if begin != end else begin
         return results_data
+
+    def _search_fetch(self, search_detail, search, limit, order):
+        results, count = super()._search_fetch(search_detail, search, limit, order)
+        query = sql.SQL("""
```

Comment:
Can't the list of event ids that match `city` be obtained through a sudo `search` instead of executing an explicit SQL query ?
Then this list could be used in the second search just in the same way as the current one.